### PR TITLE
Edit invalid index message

### LINKED ISSUE #174: [edit_student] Wrong error message when using index 0
### Current Behavior

**Command Executed**:

```
edit_student 0 n/New
```

**Actual Output**:

```
Invalid command format!
edit_student: Edits the details of the student identified by the index number used in the displayed student list. Existing values will be overwritten by the input values.
Parameters: INDEX (must be a positive integer) [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS]...
Example: edit_student 1 p/91234567 e/johndoe@example.com
```

**Observed Behavior**:

- System shows generic "Invalid command format" message
- Error message doesn't specifically mention that index must be ≥1
- Help text is shown but doesn't address the specific error

---

### Expected Behavior

**Expected Output**:

```
Error: Index must be a positive integer (1 or greater)
```

**Requirements**:

- Should specifically reject index 0 with clear error
- Message should match parameter validation in UG ("must be a positive integer")
- Should appear immediately without showing full command format

### Reproduction Steps

1. Have at least 1 student in the list
2. Execute: `edit_student 0 n/New`
3. Observe incorrect generic error message

### Impact Analysis

- **User Experience**: Confusing when learning command syntax
- **Consistency**: Differs from other commands that properly validate indices
- **Documentation**: Contradicts UG which states "positive integer"

### Suggested Fix

1. Modify `ParserUtil.parseIndex()` to explicitly check for zero:

```
public static int parseIndex(String oneBasedIndex) throws ParseException {
    // ...
    if (index == 0) {
        throw new ParseException("Index must be a positive integer (1 or greater)");
    }
    // ...
}
```

1. Update error messages to be more specific about index requirements

### Test Cases

1. Boundary cases:
    - `edit_student 0 n/Test` → Specific error
    - `edit_student 1 n/Test` → Success
    - `edit_student -1 n/Test` → Specific error

### PR DESCRIPTION
**Pull request by nicholasohjj**
Resolves #174

### CODE REVIEW

**Comment by nicholasohjj** [comment]
Resolves #194

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
  */
 public class ParserUtil {
 
-    public static final String MESSAGE_INVALID_INDEX = "Index is not a non-zero unsigned integer.";
+    public static final String MESSAGE_INVALID_INDEX = "Error: Index must be a positive integer (1 or greater)";
```
Comment (GZTan23) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:25]: > Error: ...

I think this part should be removed to be consistent with other error messages

**Review by GZTan23 — CHANGES_REQUESTED** [review]
Everything else looks good

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (nicholasohjj) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:25]: To be standardised in in future PRs? new message is more user-friendly

Comment (GZTan23) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:25]: Can 👍 Need to convey to the group

**Review by GZTan23 — APPROVED** [review]
